### PR TITLE
update subnet type deprecation source: https://docs.aws.amazon.com/cd…

### PR DESCRIPTION
**Pull request by naomatheus**
The Subnet type in our CDK infrastructure code has been deprecated.
The warning blocks deployment to staging.
Merging in with the new/correct subnet type `PRIVATE_WITH_NAT` from CDK warnings.

### CODE REVIEW

**Review by leothomas — APPROVED** [review]
Definitely try to deploy this on a staging stack first - but I don't anticipate there being any issues, I think it's the just construct changing but not the underlying architcture

**Comment by naomatheus** [comment]
I'm going to hold off on deploying this.
The staging deployment blocker seemed to point out this issue with Subnet Type. But after looking into it the warning itself may have been for an older version of CDK. 
Since I tagged you @sunu , please hold off before merging this one.

**Comment by naomatheus** [comment]
The subnet type in the CDK warning appears to also have been deprecated.
https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_ec2/SubnetType.html#aws_cdk.aws_ec2.SubnetType.PRIVATE_WITH_NAT

**Comment by naomatheus** [comment]
Changes to the deployment script should allow this deployment to pass the blocker. Will deploy shortly.

**Pull request merged**
